geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java

```diff
@@ -92,13 +92,282 @@ public static void tearDown() {
+
+    String result = jedis.get(key);
+    assertThat(result).isNull();
+
+    jedis.set(key, value);
+    result = jedis.get(key);
+    assertThat(result).isEqualTo(value);
+  }
+
+  @Test
+  public void testSET_shouldSetStringValueToKey_givenKeyContainingOtherDataType() {
+    String key = "key";
+    String stringValue = "value";
+
+    jedis.sadd(key, "member1", "member2");
+
+    jedis.set(key, stringValue);
+    String result = jedis.get(key);
+
+    assertThat(result).isEqualTo(stringValue);
+  }
+
+  @Test
+  public void testSET_shouldHonorNX_evenIfKeyContainsOtherDataType() {
+    String key = "key";
+    String stringValue = "value";
+
+    jedis.sadd(key, "member1", "member2");
+    SetParams setParams = new SetParams();
+    setParams.nx();
+
+    String result = jedis.set(key, stringValue, setParams);
+    assertThat(result).isNull();
+  }
+
+  @Test
+  public void testSET_shouldHonorXX_evenIfKeyContainsOtherDataType() {
+    String key = "key";
+    String stringValue = "value";
+
+    jedis.sadd(key, "member1", "member2");
+    SetParams setParams = new SetParams();
+    setParams.xx();
+
+    jedis.set(key, stringValue, setParams);
+    String result = jedis.get(key);
+
+    assertThat(result).isEqualTo(stringValue);
+  }
+
+  @Test
+  public void testSET_shouldAccept_EX_argumentToSetExpireTime() {
+    String key = "key";
+    String value = "value";
+    int secondsUntilExpiration = 20;
+
+    SetParams setParams = new SetParams();
+    setParams.ex(secondsUntilExpiration);
+
+    jedis.set(key, value, setParams);
+
+    Long result = jedis.ttl(key);
+
+    assertThat(result).isGreaterThan(15l);
+  }
+
+  @Test
+  public void testSET_shouldAccept_PX_argumentToSetExpireTime() {
+    String key = "key";
+    String value = "value";
+    int millisecondsUntilExpiration = 20000;
+
+    SetParams setParams = new SetParams();
+    setParams.px(millisecondsUntilExpiration);
+
+    jedis.set(key, value, setParams);
+
+    Long result = jedis.ttl(key);
+
+    assertThat(result).isGreaterThan(15l);
+  }
+
+  @Test
+  public void testSET_shouldClearPreviousTTL_onSuccess() {
+    String key = "key";
+    String value = "value";
+    int secondsUntilExpiration = 20;
+
+    SetParams setParams = new SetParams();
+    setParams.ex(secondsUntilExpiration);
+
+    jedis.set(key, value, setParams);
+
+    jedis.set(key, "other value");
+
+    Long result = jedis.ttl(key);
+
+    assertThat(result).isEqualTo(-1L);
+  }
+
+  @Test
+  public void testSET_shouldClearPreviousTTL_onXXSuccess() {
+    String key = "xx_key";
+    String value = "did exist";
+    int secondsUntilExpiration = 20;
+    SetParams setParamsXX = new SetParams();
+    setParamsXX.xx();
+    SetParams setParamsEX = new SetParams();
+    setParamsEX.ex(secondsUntilExpiration);
+    String result_EX = jedis.set(key, value, setParamsEX);
+    assertThat(result_EX).isEqualTo("OK");
+    assertThat(jedis.ttl(key)).isGreaterThan(15L);
+    jedis.set(key, value);
```

Comment:
Setting a key here seems unnecessary since the setting with XX  on line 215 should clear TTL.